app/services/account_reset/grant_request.rb

```diff
@@ -7,14 +7,16 @@ def initialize(user)
     def call
       token = SecureRandom.uuid
       arr = AccountResetRequest.find_by(user_id: @user_id)
-      arr.with_lock do
-        return false if arr.granted_token_valid?
-        account_reset_request.update(
-          granted_at: Time.zone.now,
-          granted_token: token,
-        )
```

Comment:
As of Rails 7, returning in a transaction will rollback, which is not the intended behavior in this method, though it does not affect it since there is no prior change we'd want applied.

The new [Rails/TransactionExitStatement](https://docs.rubocop.org/rubocop-rails/cops_rails.html#railstransactionexitstatement) caught it.